lib/Models/DiffableWebMapServiceCatalogItem.ts

```diff
@@ -0,0 +1,207 @@
+import { computed, runInAction } from "mobx";
+import JulianDate from "terriajs-cesium/Source/Core/JulianDate";
+import filterOutUndefined from "../Core/filterOutUndefined";
+import DiffableMixin from "../ModelMixins/DiffableMixin";
+import DiffableWebMapServiceCatalogItemTraits from "../Traits/DiffableWebMapServiceCatalogItemTraits";
+import LegendTraits from "../Traits/LegendTraits";
+import CommonStrata from "./CommonStrata";
+import createStratumInstance from "./createStratumInstance";
+import ExtendModel from "./ExtendModel";
+import LoadableStratum from "./LoadableStratum";
+import { ImageryParts } from "./Mappable";
+import { BaseModel } from "./Model";
+import { SelectableDimension } from "./SelectableDimensions";
+import WebMapServiceCatalogItem from "./WebMapServiceCatalogItem";
+import URI from "urijs";
+
+const dateFormat = require("dateformat");
+
+class DiffStratum extends LoadableStratum(
+  DiffableWebMapServiceCatalogItemTraits
+) {
+  constructor(readonly catalogItem: DiffableWebMapServiceCatalogItem) {
+    super();
+  }
+
+  duplicateLoadableStratum(model: BaseModel): this {
+    return new DiffStratum(model as DiffableWebMapServiceCatalogItem) as this;
+  }
+
+  @computed
+  get legends() {
+    if (this.catalogItem.isShowingDiff && this.diffLegendUrl) {
+      let urlMimeType;
+      try {
+        urlMimeType =
+          new URL(this.diffLegendUrl).searchParams.get("format") || undefined;
+      } catch {}
+      return [
+        createStratumInstance(LegendTraits, {
+          url: this.diffLegendUrl,
+          urlMimeType
+        })
+      ];
+    }
+    return undefined;
+  }
+
+  @computed
+  get diffLegendUrl() {
+    const diffStyleId = this.catalogItem.diffStyleId;
+    const firstDate = this.catalogItem.firstDiffDate;
+    const secondDate = this.catalogItem.secondDiffDate;
+    if (diffStyleId && firstDate && secondDate) {
+      return this.catalogItem.getLegendUrlForStyle(
+        diffStyleId,
+        JulianDate.fromIso8601(firstDate),
+        JulianDate.fromIso8601(secondDate)
+      );
+    }
+    return undefined;
+  }
+
+  @computed
+  get disableDateTimeSelector() {
+    return this.catalogItem.isShowingDiff;
+  }
+
+  @computed
+  get shortReport() {
+    const catalogItem = this.catalogItem;
+    console.log("**here**", catalogItem.isShowingDiff);
+    if (catalogItem.isShowingDiff) {
+      const format = "yyyy/mm/dd";
+      const d1 = dateFormat(catalogItem.firstDiffDate, format);
+      const d2 = dateFormat(catalogItem.secondDiffDate, format);
+      return `Showing difference image computed for ${catalogItem.diffStyleId} style on dates ${d1} and ${d2}`;
+    }
+  }
+}
+
+/**
+ * A custom WMS catalog item that supports diffing imagery from two dates.
+ */
+export default class DiffableWebMapServiceCatalogItem extends DiffableMixin(
+  ExtendModel(WebMapServiceCatalogItem, DiffableWebMapServiceCatalogItemTraits)
+) {
+  static readonly type = "diffable-wms";
```

Comment:
So does this mean if adding from a catalog json that you need to do something like 

````
{
   "type": "diffable-wms",
   "url": "http://something",
   "name": "my diffable wms"
}
````

Reply:
Yeah that's right. Sorry I didn't make that clear :grimacing: